flang/include/flang/Evaluate/tools.h

```diff
@@ -448,12 +448,12 @@ struct ExtractSubstringHelper {
 
   template <typename T>
   static std::optional<Substring> visit(const Designator<T> &e) {
-    return std::visit([](auto &&s) { return visit(s); }, e.u);
+    return Fortran::common::visit([](auto &&s) { return visit(s); }, e.u);
```

Comment:
The `Fortran::` namespace is redundant here, so in Evaluate and Semantics I don't use it.

Reply:
Done